### PR TITLE
Fix #324

### PR DESCRIPTION
**Pull request by archanaserver**
Reference:

https://github.com/theforeman/jenkins-jobs/pull/324#discussion_r1150722308

### CODE REVIEW

File: theforeman.org/pipelines/release/source/smart-proxy.groovy [hunk 1]
```diff
@@ -26,7 +26,7 @@ pipeline {
                     }
                     stage("Test Ruby") {
                         steps {
-                            run_test(ruby)
+                            run_test(env.ruby)
```
Comment (ehelms) [inline theforeman.org/pipelines/release/source/smart-proxy.groovy:29]:
```suggestion
                            run_test(ruby: env.ruby)
```

Testing this change live -- https://ci.theforeman.org/blue/organizations/jenkins/smart-proxy-develop-source-release/detail/smart-proxy-develop-source-release/178/pipeline

Comment (archanaserver) [reply theforeman.org/pipelines/release/source/smart-proxy.groovy]: @ehelms Updated this change, but seems like I need to rebase it also.

> Testing this change live -- https://ci.theforeman.org/blue/organizations/jenkins/smart-proxy-develop-source-release/detail/smart-proxy-develop-source-release/178/pipeline

**Comment by ehelms** [comment]
After some testing, this also needs this change for the job to pass:

```
                 dir(project_name) {
                     git url: git_url, branch: git_ref
+
+                    script {
+                        archive_git_hash()
+                    }
                 }
+
                 script {
-                    archive_git_hash()
                     sourcefile_paths = generate_sourcefiles(project_name: project_name, source_type: source_type)
                 }
             }
         }
     }
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: theforeman.org/pipelines/release/source/smart-proxy.groovy [hunk 2]
```diff
@@ -38,8 +38,7 @@ pipeline {
                     git url: git_url, branch: git_ref
                 }
                 script {
-                    archive_git_hash()
```
[the remark this answers is not in the document]
Comment (ehelms) [reply theforeman.org/pipelines/release/source/smart-proxy.groovy:41]: Exactly. We want:

```
                dir(project_name) {
                    git url: git_url, branch: git_ref

                    script {
                        archive_git_hash()
                    }
                }
```

File: theforeman.org/pipelines/release/source/smart-proxy.groovy [hunk 3]
```diff
@@ -36,10 +36,10 @@ pipeline {
             steps {
                 dir(project_name) {
                     git url: git_url, branch: git_ref
-                }
-                script {
-                    archive_git_hash()
-                    sourcefile_paths = generate_sourcefiles(project_name: project_name, source_type: source_type)
```
Comment (ehelms) [inline theforeman.org/pipelines/release/source/smart-proxy.groovy:42]: We still need this line. Here is what it should look like:

```
                 dir(project_name) {
                     git url: git_url, branch: git_ref

                    script {
                        archive_git_hash()
                    }
                 }

                 script {
                     sourcefile_paths = generate_sourcefiles(project_name: project_name, source_type: source_type)
                 }
```

Comment (archanaserver) [reply theforeman.org/pipelines/release/source/smart-proxy.groovy:42]: Okay, sorry I got little confuse about it. I though we can do it at the same place.

**Review by ekohl — APPROVED** [review]

**Pull request merged**